bundles/org.openhab.binding.oceanic/src/main/java/org/openhab/binding/oceanic/internal/OceanicHandlerFactory.java

```diff
@@ -40,17 +44,24 @@ public class OceanicHandlerFactory extends BaseThingHandlerFactory {
     private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections
             .unmodifiableSet(Stream.of(THING_TYPE_SERIAL, THING_TYPE_NETWORK).collect(Collectors.toSet()));
 
+    private final SerialPortManager serialPortManager;
+
+    @Activate
+    public OceanicHandlerFactory(final @Reference SerialPortManager serialPortManager) {
+        this.serialPortManager = serialPortManager;
+    }
+
     @Override
-    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+    public boolean supportsThingType(@NonNull ThingTypeUID thingTypeUID) {
         return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
     }
 
     @Override
-    protected ThingHandler createHandler(Thing thing) {
+    protected ThingHandler createHandler(@NonNull Thing thing) {
```

Comment:
The `@NonNull` annotation should not be used, see:

https://www.openhab.org/docs/developer/guidelines.html#null-annotations

```suggestion
    protected ThingHandler createHandler(Thing thing) {
```

Reply:
@wborn I reverted these changes. In fact, I started to include them because VS Code is complaining quite a lot about missing annotations. As we speak I just noticed that openHAB code formatting guidelines url (https://raw.githubusercontent.com/openhab/static-code-analysis/master/codestyle/src/main/resources/openhab_codestyle.xmi) referenced at https://www.openhab.org/docs/developer/ide/vscode.html is not working anymore